pandas/tests/series/test_operators.py

```diff
@@ -198,11 +199,14 @@ def test_scalar_na_logical_ops_corners(self):
         pytest.param(ops.ror_,
                      marks=pytest.mark.xfail(reason="GH#22092 Index "
                                                     "implementation raises",
-                                             raises=ValueError, strict=True)),
+                                             raises=InvalidIndexError,
+                                             strict=True)),
         pytest.param(ops.rxor,
                      marks=pytest.mark.xfail(reason="GH#22092 Index "
-                                                    "implementation raises",
-                                             raises=TypeError, strict=True))
+                                                    "implementation returns "
+                                                    "Index",
+                                             raises=AssertionError,
+                                             strict=True))
```

Comment:
Is this planned to be fixed before we release 0.24.0 ? 
Because we should never raise AssertionError to users

Reply:
@jorisvandenbossche any objections to this change? We're clearly in buggy territory here, but I think the remaining bugs are at least no worse than master.

Reply:
Isn't an error (although without good error message) better than a silently wrong output?

(but fine to open an issue about it for handling it later)

Reply:
I don't think the values are incorrect. I think the issue is that `xor(Series, Index)` is a Series, when apparently(?) it's supposed to be an index. If that's right, the values are correct but we're picking the wrong box to put them inside.

git blame points to https://github.com/pandas-dev/pandas/pull/23029/files (cc @jbrockmendel). Does that sound right?

Reply:
Oh sorry... the values are incorrect. OK I'll take a look at this then.

Reply:
IIRC the issue is that `xor` for Series is a different op than it is for `Index`, and so this operation is not commutative.  Usually when we have op(Index, Series) we have the Index op return `NotImplemented` and end up returning a Series, but it isn't clear that is the right thing to do here.

Reply:
Agreed that the set-like vs. array-like behavior of Index is making this a mess, but isn't, e.g. `ops.rand_(a, b)` clearly supposed to be the same as `b.__and__(a)`. There's no ambiguity there right?

Reply:
agreed